modules/azure/function_app_linux_managed_identity/main.tf

```
@@ -4,7 +4,7 @@ terraform {
   required_providers {
     azurerm = {
       source  = "hashicorp/azurerm"
-      version = "~> 3.40"
+      version = "=3.47.0"
```

```suggestion
      version = "~> 3.47"
```

Don't use a fixed version.